src/System.CommandLine/Binding/ExpressionExtensions.cs

```diff
@@ -7,30 +7,26 @@ internal static class ExpressionExtensions
     {
         internal static (Type memberType, string memberName) MemberTypeAndName<T, TValue>(this Expression<Func<T, TValue>> expression)
         {
-            if (expression == null)
+            if (expression is null)
             {
                 throw new ArgumentNullException(nameof(expression));
             }
 
             if (expression.Body is MemberExpression memberExpression)
             {
-                return TypeAndName();
+                return TypeAndName(memberExpression);
             }
 
             // when the return type of the expression is a value type, it contains a call to Convert, resulting in boxing, so we get a UnaryExpression instead
-            if (expression.Body is UnaryExpression unaryExpression)
+            if (expression.Body is UnaryExpression unaryExpression &&
```

Comment:
nit: could go one step further with recursive patterns: 
`if (expression.Body is UnaryExpression { Operand: MemberExpression operandMemberExpression } _)`